### PR TITLE
[v1.x] [Backport] Add checks in batchnorm's infer shape

### PR DESCRIPTION
**Pull request by bgawrych**
## Description ##
Add checks to infer shape to avoid passing invalid tensors. Connected to issue: #19065

Backport of: https://github.com/apache/incubator-mxnet/pull/20415

### CODE REVIEW

**Comment by szha** [comment]
@mxnet-bot run ci [all]

**Comment by bgawrych** [comment]
@mxnet-bot run ci [centos-gpu, unix-gpu]

**Review by bartekkuncer — CHANGES_REQUESTED** [review]
Please apply formatter to the changed code.

**Comment by bgawrych** [comment]
> Please apply formatter to the changed code.
Done for my changes - for whole files (src/operator/nn/batch_norm.cc  and tests/cpp/operator/batchnorm_test.cc) probably would be better to make it in other commit with more formatting changes

**Comment by bgawrych** [comment]
@mxnet-bot run ci [unix-gpu]

**Review by bartekkuncer — APPROVED** [review]
LGTM.

**Review by akarbown — APPROVED** [review]
